packages/vscode-extension/lib/runtime.js

```diff
@@ -25,15 +25,74 @@ global.__RNIDE_onDebuggerReady = function () {
 // debug adapter and avoid exposing as part of application logs
 console.log("__RNIDE_INTERNAL", "radon-ide runtime loaded");
 
+let stackOffset = 1; // default offset is 1, because the first frame is the wrapConsole function
+let logRef = null;
+let errorRef = null;
+let warnRef = null;
+let infoRef = null;
+
+function isConsoleRefChanged() {
+  return (
+    logRef !== console.log ||
+    errorRef !== console.error ||
+    warnRef !== console.warn ||
+    infoRef !== console.info
+  );
+}
+
+function updateConsoleRefs() {
+  logRef = console.log;
+  warnRef = console.warn;
+  errorRef = console.error;
+  infoRef = console.info;
+}
+
 function wrapConsole(consoleFunc) {
+  let initializationStack = [];
+  let sendInitialLog = false;
+
   return function (...args) {
     const stack = parseErrorStack(new Error().stack);
-    const expoLogIndex = stack.findIndex((frame) => frame.methodName === "__expoConsoleLog");
-    const location = expoLogIndex > 0 ? stack[expoLogIndex + 1] : stack[1];
+
+    // To get the proper stack frame, so we can display link to the source code
+    // we need to skip wrappers (like wrapConsole below or for example Sentry wrapper)
+    // Otherwise, the stack frame would point to the wrapper and not the actual source code
+    // To do that, we run console.log again in runWrapper, and then compare
+    // first frames to find the offset. We do that when ant of console ref changes
+    if (isConsoleRefChanged()) {
+      if (sendInitialLog) {
+        sendInitialLog = false;
+
+        for (let i = 0; i < Math.min(stack.length, initializationStack.length); i++) {
+          const diffLine = stack[i].lineNumber !== initializationStack[i].lineNumber;
+          const diffColumn = stack[i].column !== initializationStack[i].column;
+
+          if (diffLine || diffColumn) {
+            // Repeat the sampling when stackOffset changes
+            if (i !== stackOffset) {
+              sampleCount = 0;
+            }
+
+            stackOffset = i;
+            break;
+          }
+        }
+
+        updateConsoleRefs();
+        return;
+      }
+
+      initializationStack = stack;
+      sendInitialLog = true;
+      console.log(); // Recursive dummy console.log to get the source maps
```

Comment:
Woudl be much cleaner if we set flag immediately before and reset immediately after
```
logFunctionReentryFlag = true;
console.log();
logFunctionReentryFlag = false;

// calculate new offset here
```


when log function is called and `logFunctionReentryFlag` is set, the only thing we do, is we save current stack and exit to keep things simple:

```
let logFunctionReentryStack = null;
let logFunctionReentryFlag = false;

return function (...args) {
    const stack = parseErrorStack(new Error().stack);
   
    if (logFunctionReendrtFlag) {
        logFunctionReentryStack = stack;
        return;
    }
```

Reply:
Yeah, I think that looks much cleaner, thanks!